### ISSUE TITLE
Support more xvega functionalities

### ISSUE THREAD

**Issue by marimeireles** [jupyter-xeus/xeus-sqlite#62]
If you have a look at very cool's @madhur-tandon work in [xvega](https://github.com/Quantstack/xvega), specifically if you run the [binder](https://mybinder.org/v2/gh/QuantStack/xvega/stable?urlpath=lab%2Ftree%2Fnotebooks%2Fdemo.ipynb) you'll see that xeus-sqlite is not supporting quite a few of those.
This issue is an effort to list them and try to give some pointers on how to do it. If you're getting started on C++ and would like to learn in a real project this is a very good opportunity! The PRs to resolve this issue are going to be self contained and this is a small project, so it's somewhat easy to get a grasp of everything that's going on here. Besides, this is a mentored issue, which means you can feel comfortable to ask whatever questions you might have + all maintainers of this repo are under the NumFocus code of conduct and it's a safe space.

- [ ]  color
- [ ]  field type (quantitative, temporal, etc)
- [ ] different kinds of marks (line, area, etc)
- [ ] field aggregate attribute
- [ ]  layer
- [ ]  scale
- [ ]  direction
- [ ]  config

In order to add these functionalities to `xeus-sqlite` you'll need to get somewhat familiar with `xvega` code.

Here's my recommended step to step to get a contribution going:
1. Make sure you're able to build both projects locally, xvega and xeus-sqlite
2. Test both project's notebooks and get familiar with basic functionalities
3. Choose a functionality of this list and search for it on the xvega code. Read through it and the functions that are using it, the data structures around it, etc. Just so you get a first initial idea. (Don't worry if it doesn't make a lot of sense, it's the first time you're seeing this!)
4. Read the `xvega_bindings.cpp` file on xeus-sqlite and see how the other functionalities were already integrated.
5. Kind of copying the pattern that you saw on 4, give it a try on implementing yours! :)

If you have any questions, don't hesitate to ask in our chat: https://gitter.im/QuantStack/Lobby
Someone will answer you ASAP :D

**Comment by MooersLab** [comment]
Yes, support for layers would be useful for plotting to series of Y-values on the same plot.
